includes/06-ajax.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * WordPress ajax handlers
+ *
+ * @package ubc_wp_vote
+ */
+
+namespace UBC\CTLT\WPVote\Ajax;
+
+// If this file is called directly, abort.
+if ( ! defined( 'ABSPATH' ) ) {
+	die;
+}
+
+// Hooks.
+add_action( 'wp_ajax_ubc_ctlt_wp_vote_upvote', __NAMESPACE__ . '\\ubc_ctlt_wp_vote_upvote' );
+add_action( 'wp_ajax_ubc_ctlt_wp_vote_downvote', __NAMESPACE__ . '\\ubc_ctlt_wp_vote_downvote' );\
+add_action( 'wp_ajax_ubc_ctlt_wp_vote_rating', __NAMESPACE__ . '\\ubc_ctlt_wp_vote_rating' );
+
+/**
+ * Ajax handler for upvote toggle action.
+ *
+ * @return void
+ */
+function ubc_ctlt_wp_vote_upvote() {
+	check_ajax_referer( 'ubc', 'security' );
+
+	$object_id = isset( $_POST['object_id'] ) ? intval( $_POST['object_id'] ) : null;
+	$object_type = isset( $_POST['object_type'] ) ? sanitize_text_field( wp_unslash( $_POST['object_type'] ) ) : null;
+
+	$success = \UBC\CTLT\WPVote\WP_Vote::do_current_user_up_vote(
+		array(
+			'object_id' => $object_id,
+			'object_type' => $object_type,
+		)
+	);
+
+	if ( false === $success ) {
+		wp_die( esc_html( __( 'System Error' ) ), 'Error', array( 'response' => 400 ) );
+	}
+
+	wp_die();
+}//end ubc_ctlt_wp_vote_upvote()
+
+/**
+ * Ajax handler for downvote toggle action.
+ *
+ * @return void
+ */
+function ubc_ctlt_wp_vote_downvote() {
+	check_ajax_referer( 'ubc', 'security' );
+
+	$object_id = isset( $_POST['object_id'] ) ? intval( $_POST['object_id'] ) : null;
+	$object_type = isset( $_POST['object_type'] ) ? sanitize_text_field( wp_unslash( $_POST['object_type'] ) ) : null;
+
+	$success = \UBC\CTLT\WPVote\WP_Vote::do_current_user_down_vote(
+		array(
+			'object_id' => $object_id,
+			'object_type' => $object_type,
+		)
+	);
+
+	if ( false === $success ) {
+		wp_die( esc_html( __( 'System Error' ) ), 'Error', array( 'response' => 400 ) );
```

Comment:
Let's have different 'Error' values for different errors. Rather than 'System Error' for all of the messages. Let's have an error message that is related to the function it is in, and the type of error that is happening. i.e. here perhaps we could have "Error in 'ubc_ctlt_wp_vote_downvote()'. 'do_current_user_down_vote()' returned false. Object ID: $object_id, Object Type: $object_type"